### PR TITLE
Used C++ style cast instead of C style cast 

### PR DESCRIPTION
**Pull request by abhijelly**
### Description

Please explain the changes you made, including a reference to the related issue if applicable

### Checklist
- [ ] **Required by CI**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extend the tutorials / documentation [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Document API changes relevant to the user in the [MIGRATION.md](https://github.com/ros-planning/moveit/blob/master/MIGRATION.md) notes
- [ ] Create tests, which fail without this PR [reference](https://ros-planning.github.io/moveit_tutorials/doc/tests/tests_tutorial.html)
- [ ] Include a screenshot if changing a GUI
- [ ] While waiting for someone to review your request, please help review [another open pull request](https://github.com/ros-planning/moveit/pulls) to support the maintainers

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_core/collision_distance_field/src/collision_distance_field_types.cpp [hunk 1]
```diff
@@ -451,7 +451,7 @@ void getProximityGradientMarkers(const std::string& frame_id, const std::string&
   if (gradients.size() != posed_decompositions.size() + posed_vector_decompositions.size())
   {
     RCLCPP_WARN(LOGGER, "Size mismatch between gradients %u and decompositions %u", (unsigned int)gradients.size(),
```
Comment (Abishalini) [inline moveit_core/collision_distance_field/src/collision_distance_field_types.cpp:453]:
```suggestion
    RCLCPP_WARN(LOGGER, "Size mismatch between gradients %u and decompositions %u", static_cast<unsigned int>(gradients.size()),
```

Comment (abhijelly) [reply moveit_core/collision_distance_field/src/collision_distance_field_types.cpp]: Done!

**Review by henningkayser — COMMENTED** [review]
Thank you for this cleanup. All casts look correct to me, just some comments before approval.

**Comment by abhijelly** [comment]
> Thank you for this cleanup. All casts look correct to me, just some comments before approval.

Done!

**Review by henningkayser — APPROVED** [review]

**Review by tylerjw — APPROVED** [review]

**Pull request merged**